src/platforms/windows/daemon/windowstunnelservice.cpp

```diff
@@ -162,56 +97,27 @@ void WindowsTunnelService::timeout() {
   emit backendFailure();
 }
 
-int WindowsTunnelService::nextLogIndex() {
-  if (!m_logdata) {
-    return 0;
-  }
-  qint32 value;
-  memcpy(&value, m_logdata + RINGLOG_INDEX_OFFSET, 4);
-  return value % RINGLOG_MAX_ENTRIES;
-}
-
-void WindowsTunnelService::processMessage(int index) {
-  Q_ASSERT(index >= 0);
-  Q_ASSERT(index < RINGLOG_MAX_ENTRIES);
-  size_t offset = index * (RINGLOG_TIMESTAMP_SIZE + RINGLOG_MESSAGE_SIZE);
-  uchar* data = m_logdata + RINGLOG_HEADER_SIZE + offset;
-
-  quint64 timestamp;
-  memcpy(&timestamp, data, 8);
-  if (timestamp <= m_logEpochNsec) {
-    return;
-  }
-
-  QByteArray message((const char*)data + RINGLOG_TIMESTAMP_SIZE,
-                     RINGLOG_MESSAGE_SIZE);
-  logdll.info() << QString::fromUtf8(message);
-}
-
-void WindowsTunnelService::processLogs() {
-  /* On the first pass, scan all log messages. */
-  if (m_logindex < 0) {
-    m_logindex = nextLogIndex();
-    processMessage(m_logindex);
-    m_logindex = (m_logindex + 1) % RINGLOG_MAX_ENTRIES;
-  }
+bool WindowsTunnelService::start(const QString& configFile) {
+  logger.debug() << "Starting the tunnel service";
 
-  /* Report new messages. */
-  while (m_logindex != nextLogIndex()) {
-    processMessage(m_logindex);
-    m_logindex = (m_logindex + 1) % RINGLOG_MAX_ENTRIES;
+  m_logworker = new WindowsTunnelLogger(WindowsCommons::tunnelLogFile());
+  if (m_logworker) {
```

Comment:
`new` doesn't fail. remove this `if (m_logworker) {`

Reply:
Ah, the joy of virtual memory. I am far too used to treating memory allocation with suspicion :)